command/build.go

```diff
@@ -156,6 +88,12 @@ func (c *BuildCommand) RunContext(buildCtx context.Context, cla *BuildArgs) int
 		return ret
 	}
 
+	diags = SetupRegistry(packerStarter)
+	ret = writeDiags(c.Ui, nil, diags)
+	if ret != 0 {
+		return ret
+	}
+
```

Comment:
Should this be guarded only setup if needed.

Reply:
Sorry this was my leaving a comment for myself and I submitted it... 😜

Reply:
Since the checks are done within the function, I'd imagine we can leave it as-is, unless you think it'd be better to do the checks here too?
We can't rely on `ConfiguredArtifactMetadataPublisher` here too, since the `Bucket` is only set in the `SetupRegistry` function.

Reply:
Yeah I think it is fine to encapsulate the logic into the function as you have it. I think the name is what through me. Seeing a call to SetupRegistry in the build command at first glance reads like it is always configured, and part of the build.

Reply:
I see, should we rename the function to `TrySetupHCP`? Do you think it'd help clear the confusion?